databroker/src/grpc/kuksa_val_v2/val.rs

```diff
@@ -39,12 +39,42 @@ const MAX_REQUEST_PATH_LENGTH: usize = 1000;
 impl proto::val_server::Val for broker::DataBroker {
     async fn get_value(
         &self,
-        _request: tonic::Request<proto::GetValueRequest>,
+        request: tonic::Request<proto::GetValueRequest>,
     ) -> Result<tonic::Response<proto::GetValueResponse>, tonic::Status> {
-        Err(tonic::Status::new(
-            tonic::Code::Unimplemented,
-            "Unimplemented",
-        ))
+        debug!(?request);
+        let permissions = match request.extensions().get::<Permissions>() {
+            Some(permissions) => {
+                debug!(?permissions);
+                permissions.clone()
+            }
+            None => return Err(tonic::Status::unauthenticated("Unauthenticated")),
+        };
+
+        let broker = self.authorized_access(&permissions);
+
+        let request = request.into_inner();
+
+        let signal_id = match get_signal(request.signal_id, &broker).await {
+            Ok(signal_id) => signal_id,
+            Err(err) => return Err(err),
+        };
+
+        let datapoint = match broker.get_datapoint(signal_id).await {
+            Ok(datapoint) => datapoint,
+            Err(ReadError::NotFound) => {
+                return Err(tonic::Status::new(tonic::Code::NotFound, "Path not found"))
+            }
+            Err(ReadError::PermissionDenied | ReadError::PermissionExpired) => {
```

Comment:
shouldn't we return UNAUTHENTICATED if the permissions are expired?

Reply:
I think we have been a bit inconsistent with that in the past, but I think it makes sense. Any opinion @argerus?

Reply:
Changed it in latest commit